app/(main)/league/[leagueId]/entry/all/page.test.tsx

```diff
@@ -65,4 +85,54 @@ describe('Entry Component', () => {
       expect(screen.queryByTestId('global-spinner')).not.toBeInTheDocument();
     });
   });
+  test('should display "Make Pick" button when no pick is set', async () => {
+    mockUseDataStore.mockReturnValueOnce({ user: { id: '123', leagues: [] } });
+    mockGetGameWeek.mockResolvedValueOnce({ week: 1 });
+    mockGetCurrentUserEntries.mockResolvedValueOnce([
+      {
```

Comment:
Can you break down why we are mocking a return value for `{ user: { id: '123', leagues: [] } }`, `{ week: 1 }`, and `{    $id: '123',
        name: 'Test Entry',
        week: 1,
        selectedTeams: [],
      },` here?

Reply:
Of course :) @choir27 All of them are different values returned from the API calls called in Entry page.

1. This `{ user: { id: '123', leagues: [] } }` represents the value returned from `useDataStore` hook

3. This `{ week: 1 }` mocks the value returned by the `getGameWeek` API call

4. This `{ $id: '123', name: 'Test Entry', week: 1, selectedTeams: [], }` is mocking the `getCurrentUserEntries` API call

I hope this helps!

Reply:
question @vmaineng: what is the purpose of mockReturnValueOnce? And why not save the mocked value `{ user: { id: '123', leagues: [] } }` in its own mock value to be reused? Instead of having to write it out multiple times. Same with `{week: 1}`.

Reply:
@jennifertieu This is a great question! I do like this idea and curious of how were you thinking of implementing it? Should it be stored into a variable? Let me know what your thoughts are :)

Reply:
@vmaineng yes, I think storing it in a variable would be better. That would also help us if we needed to update or reuse the values in the future.

Reply:
@jennifertieu thinking about it now, it's not a bad idea to store it in a variable, but what if we have a test for `{week: 2 }` and if we want to return something back for `leagues` in `{ user: { id: '123', leagues: [] } }`? Therefore, I don't mind being explicit like this for the tests. What are your thoughts? :)